### ISSUE TITLE
How to copy range value from one roaringbitmap to another object 's offset range?

### ISSUE THREAD

**Issue by brucenan** [RoaringBitmap/RoaringBitmap#331]
Is there any high performance method to copy a range(int x, int y) of RoaringBitmap to another Roaringbitmap object 's position (x+offset, y+offset)?

Now I copy it one by one bit in a loop, it seems too slow.

I found that RoaringBItmap.addoffset seems has the similar function ,but the parameter offset only be more than zero, that mean i can not move it to low position

**Comment by lemire** [comment]
@brucenan 
Something like the following should do?
```java
y.or(new RoaringBitmap().addRange(start,end).and(x)));
```

**Comment by brucenan** [comment]
> ```java
> new RoaringBitmap().addRange(start,end).and(x)
> ```

Thanks for your feedback. 
But it seems that y still has a range at position(start, end). I want a y object that has range from (0, end-start+1) position, like a left shit operation.

**Comment by lemire** [comment]
```java
y.or(RoaringBitmap.addOffset(new RoaringBitmap().addRange(start,end).and(x), offset));
```

**Comment by brucenan** [comment]
> ```java
> y.or(RoaringBitmap.addOffset(new RoaringBitmap().addRange(start,end).and(x), offset));
> ```

Thanks. Seems you've already reply in another issue, offset is only positive integer now. 
Waiting for the next release for negative offset !

**Comment by lemire** [comment]
We may allow negative offsets in the near future:
https://github.com/RoaringBitmap/RoaringBitmap/pull/332

**Comment by lemire** [comment]
> what about the performance of this offset operation?

Shifting the data by arbitrary offsets is potentially expensive even with a good support from the library.  It means copying and moving lots of data. It is best if you can avoid it.

This being said, though I have not benchmarked the addOffset function, it should be 10x faster than doing it with iterators, at least.

Still: if you are constantly using the addOffset function, then you may want to revisit your design.

**Comment by lemire** [comment]
It is now possible to use negative offsets.

**Comment by lemire** [comment]
Fixed in RoaringBitmap-0.8.5.
